src/main/java/com/ioet/bpm/people/services/PasswordManagementService.java

```diff
@@ -0,0 +1,63 @@
+package com.ioet.bpm.people.services;
+
+import org.springframework.stereotype.Component;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.util.Base64;
+
+@Component
+public class PasswordManagementService {
+
+    private static final String ALGORITHM = "SHA-512";
+    private static final int ITERATIONS = 10;
+    private static final int SALT_SIZE = 64;
+
+
+    public String generatePassword(String password) throws UnsupportedEncodingException, NoSuchAlgorithmException {
+        byte[] salt = generateSalt();
+        byte[] hash = calculateHash(password, salt);
+        return Base64.getEncoder().encodeToString(salt).concat(":").concat(Base64.getEncoder().encodeToString(hash));
+    }
+
+
+    public byte[] calculateHash(String password, byte[] salt) throws NoSuchAlgorithmException,
+            UnsupportedEncodingException {
+        MessageDigest md = MessageDigest.getInstance(ALGORITHM);
```

Comment:
Since the algorithm is well known and we know it will never throw an exception you can catch the error here and remove the `throws` statement.